<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<!--Converted with LaTeX2HTML 2018.3 (Released July 19, 2018) -->
<HTML lang="EN">
<HEAD>
<TITLE>MODELLER A Program for Protein Structure Modeling Release 9.24, r11614</TITLE>
<META NAME="description" CONTENT="MODELLER A Program for Protein Structure Modeling Release 9.24, r11614">
<META NAME="keywords" CONTENT="manual">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=utf-8">
<META NAME="viewport" CONTENT="width=device-width, initial-scale=1.0">
<META NAME="Generator" CONTENT="LaTeX2HTML v2018.3">

<LINK REL="STYLESHEET" HREF="manual.css">
<LINK REL="STYLESHEET" HREF="pygments.css">

<LINK REL="next" HREF="node1.html">
</HEAD>

<BODY >

<DIV CLASS="navigation"><!--Navigation Panel-->
<A
 HREF="node1.html">
<IMG WIDTH="37" HEIGHT="24" ALT="next" SRC="next.png"></A> 
<IMG WIDTH="26" HEIGHT="24" ALT="up" SRC="up_g.png"> 
<IMG WIDTH="63" HEIGHT="24" ALT="previous" SRC="prev_g.png"> 
<A ID="tex2html298"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALT="contents" SRC="contents.png"></A> 
<A ID="tex2html300"
  HREF="node513.html">
<IMG WIDTH="43" HEIGHT="24" ALT="index" SRC="index.png"></A> 
<BR>
<B> Next:</B> <A
 HREF="node1.html">Contents</A>
 &nbsp; <B>  <A ID="tex2html299"
  HREF="node1.html">Contents</A></B> 
 &nbsp; <B>  <A ID="tex2html301"
  HREF="node513.html">Index</A></B> 
<BR>
<BR></DIV>
<!--End of Navigation Panel-->

<P>

<P>

  <H1 class="CENTER">MODELLER 
<BR>
A Program for Protein Structure Modeling 
<BR>
Release 9.24, r11614</H1>
<P class="CENTER"><STRONG>Andrej &#352;ali</STRONG>
<BR><I>with help from</I>
</P><P class="CENTER"><STRONG>Ben Webb, M.S. Madhusudhan, Min-Yi Shen, Guangqiang Dong,</STRONG>
</P><P class="CENTER"><STRONG>Marc A. Mart&#305;-Renom, Narayanan Eswar, Frank Alber, Maya Topf,</STRONG>
</P><P class="CENTER"><STRONG>Baldomero Oliva, Andr&#225;s Fiser, Roberto S&#225;nchez, Bozidar Yerkovich,</STRONG>
</P><P class="CENTER"><STRONG>Azat Badretdinov, Francisco Melo, John P. Overington, and Eric Feyfant</STRONG>
</P><P class="CENTER"><STRONG>email: <TT>modeller-care AT salilab.org</TT></STRONG>
</P><P class="CENTER"><STRONG>URL <A ID="tex2html1"
  HREF="https://salilab.org/modeller/"><TT>https://salilab.org/modeller/</TT></A>
</STRONG>
</P>
<BR><P class="CENTER"><B>Date:</B> 2020/04/06</P>

<HR>

<P>
<BR><HR>
<!--Table of Child-Links-->
<A ID="CHILD_LINKS"></A>

<UL CLASS="ChildLinks">
<LI><A ID="tex2html302"
  HREF="node1.html">Contents</A>
<LI><A ID="tex2html303"
  HREF="node2.html">List of Figures</A>
<LI><A ID="tex2html304"
  HREF="node3.html">List of Tables</A>
<LI><A ID="tex2html305"
  HREF="node4.html">Copyright notice</A>
<LI><A ID="tex2html306"
  HREF="node5.html">Acknowledgments</A>
<LI><A ID="tex2html307"
  HREF="node6.html">Introduction</A>
<UL>
<LI><A ID="tex2html308"
  HREF="node7.html">What is M<SMALL>ODELLER</SMALL>?</A>
<LI><A ID="tex2html309"
  HREF="node8.html">M<SMALL>ODELLER</SMALL> bibliography</A>
<LI><A ID="tex2html310"
  HREF="node9.html">Obtaining and installing the program</A>
<LI><A ID="tex2html311"
  HREF="node10.html">Bug reports</A>
<LI><A ID="tex2html312"
  HREF="node11.html">Method for comparative protein structure modeling by 
M<SMALL>ODELLER</SMALL></A>
<LI><A ID="tex2html313"
  HREF="node12.html">Using M<SMALL>ODELLER</SMALL> for comparative modeling</A>
<UL>
<LI><A ID="tex2html314"
  HREF="node13.html">Preparing input files</A>
<UL>
<LI><A ID="tex2html315"
  HREF="node13.html#SECTION00661100000000000000">Atom files</A>
<LI><A ID="tex2html316"
  HREF="node13.html#SECTION00661200000000000000">Alignment file</A>
<LI><A ID="tex2html317"
  HREF="node13.html#SECTION00661300000000000000">Script file</A>
</UL>
<LI><A ID="tex2html318"
  HREF="node14.html">Running M<SMALL>ODELLER</SMALL></A>
</UL>
</UL>
<BR>
<LI><A ID="tex2html319"
  HREF="node15.html">Automated comparative modeling with <TT>automodel</TT></A>
<UL>
<LI><A ID="tex2html320"
  HREF="node16.html">Simple usage</A>
<LI><A ID="tex2html321"
  HREF="node17.html">More advanced usage</A>
<UL>
<LI><A ID="tex2html322"
  HREF="node18.html">Including water molecules, HETATM residues, and hydrogen atoms</A>
<LI><A ID="tex2html323"
  HREF="node19.html">Changing the default optimization and refinement protocol</A>
<LI><A ID="tex2html324"
  HREF="node20.html">Getting a very fast and approximate model</A>
<LI><A ID="tex2html325"
  HREF="node21.html">Building a model from multiple templates</A>
<LI><A ID="tex2html326"
  HREF="node22.html">Building an all hydrogen model</A>
<LI><A ID="tex2html327"
  HREF="node23.html">Refining only part of the model</A>
<LI><A ID="tex2html328"
  HREF="node24.html">Including disulfide bridges</A>
<LI><A ID="tex2html329"
  HREF="node25.html">Generating new-style PDBx/mmCIF outputs</A>
<LI><A ID="tex2html330"
  HREF="node26.html">Providing your own restraints file</A>
<LI><A ID="tex2html331"
  HREF="node27.html">Using your own initial model</A>
<LI><A ID="tex2html332"
  HREF="node28.html">Adding additional restraints to the defaults</A>
<LI><A ID="tex2html333"
  HREF="node29.html">Building multi-chain models</A>
<LI><A ID="tex2html334"
  HREF="node30.html">Residues and chains in multi-chain models</A>
<LI><A ID="tex2html335"
  HREF="node31.html">Accessing output data after modeling is complete</A>
<LI><A ID="tex2html336"
  HREF="node32.html">Fully automated alignment and modeling</A>
</UL>
<LI><A ID="tex2html337"
  HREF="node33.html">Loop optimization</A>
<UL>
<LI><A ID="tex2html338"
  HREF="node34.html">Automatic loop refinement after model building</A>
<LI><A ID="tex2html339"
  HREF="node35.html">Defining loop regions for refinement</A>
<LI><A ID="tex2html340"
  HREF="node36.html">Refining an existing PDB file</A>
</UL>
</UL>
<BR>
<LI><A ID="tex2html341"
  HREF="node37.html">Frequently asked questions and history</A>
<UL>
<LI><A ID="tex2html342"
  HREF="node38.html">Frequently asked questions (FAQ) and examples</A>
<LI><A ID="tex2html343"
  HREF="node39.html">M<SMALL>ODELLER</SMALL> updates</A>
<UL>
<LI><A ID="tex2html344"
  HREF="node40.html">Changes since release 9.23</A>
<LI><A ID="tex2html345"
  HREF="node41.html">Changes since release 9.22</A>
</UL>
</UL>
<BR>
<LI><A ID="tex2html346"
  HREF="node42.html">Comparative modeling class reference</A>
<UL>
<LI><A ID="tex2html347"
  HREF="node43.html"><TT>automodel</TT> reference</A>
<UL>
<LI><A ID="tex2html348"
  HREF="node44.html">automodel() &mdash; prepare to build one or more comparative models</A>
<LI><A ID="tex2html349"
  HREF="node45.html">automodel.library_schedule &mdash; select optimization schedule</A>
<LI><A ID="tex2html350"
  HREF="node46.html">automodel.md_level &mdash; control the model refinement level</A>
<LI><A ID="tex2html351"
  HREF="node47.html">automodel.outputs &mdash; all output data for generated models</A>
<LI><A ID="tex2html352"
  HREF="node48.html">automodel.rand_method &mdash; control initial model randomization</A>
<LI><A ID="tex2html353"
  HREF="node49.html">automodel.generate_method &mdash; control initial model generation</A>
<LI><A ID="tex2html354"
  HREF="node50.html">automodel.max_var_iterations &mdash; select length of optimizations</A>
<LI><A ID="tex2html355"
  HREF="node51.html">automodel.repeat_optimization &mdash; number of times to repeat optimization</A>
<LI><A ID="tex2html356"
  HREF="node52.html">automodel.max_molpdf &mdash; objective function cutoff</A>
<LI><A ID="tex2html357"
  HREF="node53.html">automodel.initial_malign3d &mdash; initial template alignment</A>
<LI><A ID="tex2html358"
  HREF="node54.html">automodel.starting_model &mdash; first model to build</A>
<LI><A ID="tex2html359"
  HREF="node55.html">automodel.ending_model &mdash; last model to build</A>
<LI><A ID="tex2html360"
  HREF="node56.html">automodel.final_malign3d &mdash; final template-model alignment</A>
<LI><A ID="tex2html361"
  HREF="node57.html">automodel.write_intermediates &mdash; write intermediate files during optimization</A>
<LI><A ID="tex2html362"
  HREF="node58.html">automodel.trace_output &mdash; control optimization output</A>
<LI><A ID="tex2html363"
  HREF="node59.html">automodel.max_ca_ca_distance &mdash; Distance cutoff for CA-CA homology-derived restraints</A>
<LI><A ID="tex2html364"
  HREF="node60.html">automodel.max_n_o_distance &mdash; Distance cutoff for N-O homology-derived restraints</A>
<LI><A ID="tex2html365"
  HREF="node61.html">automodel.max_sc_mc_distance &mdash; Distance cutoff for sidechain-mainchain homology-derived restraints</A>
<LI><A ID="tex2html366"
  HREF="node62.html">automodel.max_sc_sc_distance &mdash; Distance cutoff for sidechain-sidechain homology-derived restraints</A>
<LI><A ID="tex2html367"
  HREF="node63.html">automodel.blank_single_chain &mdash; Control chain ID for single-chain models</A>
<LI><A ID="tex2html368"
  HREF="node64.html">automodel.set_output_model_format() &mdash; set format for output models</A>
<LI><A ID="tex2html369"
  HREF="node65.html">automodel.get_optimize_actions() &mdash; get actions to carry out during the initial optimization</A>
<LI><A ID="tex2html370"
  HREF="node66.html">automodel.get_refine_actions() &mdash; get actions to carry out during the refinement</A>
<LI><A ID="tex2html371"
  HREF="node67.html">automodel.select_atoms() &mdash; select region for optimization and assessment</A>
<LI><A ID="tex2html372"
  HREF="node68.html">automodel.auto_align() &mdash; generate an automatic initial alignment</A>
<LI><A ID="tex2html373"
  HREF="node69.html">automodel.very_fast() &mdash; request rapid optimization</A>
<LI><A ID="tex2html374"
  HREF="node70.html">automodel.make() &mdash; build all models</A>
<LI><A ID="tex2html375"
  HREF="node71.html">automodel.cluster() &mdash; cluster all built models</A>
<LI><A ID="tex2html376"
  HREF="node72.html">automodel.special_restraints() &mdash; add additional restraints</A>
<LI><A ID="tex2html377"
  HREF="node73.html">automodel.nonstd_restraints() &mdash; add restraints on ligands</A>
<LI><A ID="tex2html378"
  HREF="node74.html">automodel.special_patches() &mdash; add additional patches to the topology</A>
<LI><A ID="tex2html379"
  HREF="node75.html">automodel.user_after_single_model() &mdash; analyze or refine each model</A>
<LI><A ID="tex2html380"
  HREF="node76.html">automodel.get_model_filename() &mdash; get the model PDB/mmCIF name</A>
<LI><A ID="tex2html381"
  HREF="node77.html">automodel.use_parallel_job() &mdash; parallelize model building</A>
<LI><A ID="tex2html382"
  HREF="node78.html">automodel.guess_atom_types() &mdash; automatically assign C<SMALL>HARMM</SMALL> atom types</A>
<LI><A ID="tex2html383"
  HREF="node79.html">automodel.guess_atom_type() &mdash; automatically assign C<SMALL>HARMM</SMALL> atom type</A>
</UL>
<LI><A ID="tex2html384"
  HREF="node80.html"><TT>allhmodel</TT> reference</A>
<UL>
<LI><A ID="tex2html385"
  HREF="node81.html">allhmodel() &mdash; prepare to build all-hydrogen models</A>
</UL>
<LI><A ID="tex2html386"
  HREF="node82.html"><TT>loopmodel</TT> reference</A>
<UL>
<LI><A ID="tex2html387"
  HREF="node83.html">loopmodel() &mdash; prepare to build models with loop refinement</A>
<LI><A ID="tex2html388"
  HREF="node84.html">loopmodel.loop.md_level &mdash; control the loop model refinement level</A>
<LI><A ID="tex2html389"
  HREF="node85.html">loopmodel.loop.max_var_iterations &mdash; select length of optimizations</A>
<LI><A ID="tex2html390"
  HREF="node86.html">loopmodel.loop.library_schedule &mdash; select optimization schedule</A>
<LI><A ID="tex2html391"
  HREF="node87.html">loopmodel.loop.starting_model &mdash; first loop model to build</A>
<LI><A ID="tex2html392"
  HREF="node88.html">loopmodel.loop.ending_model &mdash; last loop model to build</A>
<LI><A ID="tex2html393"
  HREF="node89.html">loopmodel.loop.write_selection_only &mdash; write PDB/mmCIFs containing only the loops</A>
<LI><A ID="tex2html394"
  HREF="node90.html">loopmodel.loop.write_defined_only &mdash; only write non-loop atoms present in the input model</A>
<LI><A ID="tex2html395"
  HREF="node91.html">loopmodel.loop.outputs &mdash; all output data for generated loop models</A>
<LI><A ID="tex2html396"
  HREF="node92.html">loopmodel.select_loop_atoms() &mdash; select region for loop optimization and assessment</A>
<LI><A ID="tex2html397"
  HREF="node93.html">loopmodel.get_loop_model_filename() &mdash; get the model PDB/mmCIF name</A>
<LI><A ID="tex2html398"
  HREF="node94.html">loopmodel.user_after_single_loop_model() &mdash; analyze or refine each loop model</A>
<LI><A ID="tex2html399"
  HREF="node95.html">loopmodel.read_potential() &mdash; read in the loop modeling potential</A>
<LI><A ID="tex2html400"
  HREF="node96.html">loopmodel.build_ini_loop() &mdash; create the initial conformation of the loop</A>
</UL>
<LI><A ID="tex2html401"
  HREF="node97.html"><TT>dope_loopmodel</TT> reference</A>
<UL>
<LI><A ID="tex2html402"
  HREF="node98.html">dope_loopmodel() &mdash; prepare to build models with DOPE loop refinement</A>
</UL>
<LI><A ID="tex2html403"
  HREF="node99.html"><TT>dopehr_loopmodel</TT> reference</A>
</UL>
<BR>
<LI><A ID="tex2html404"
  HREF="node100.html">M<SMALL>ODELLER</SMALL> general reference</A>
<UL>
<LI><A ID="tex2html405"
  HREF="node101.html">Miscellaneous rules and features of M<SMALL>ODELLER</SMALL></A>
<UL>
<LI><A ID="tex2html406"
  HREF="node102.html">M<SMALL>ODELLER</SMALL> system</A>
<LI><A ID="tex2html407"
  HREF="node103.html">Controlling breakpoints and the amount of output</A>
<LI><A ID="tex2html408"
  HREF="node104.html">File naming</A>
<UL>
<LI><A ID="tex2html409"
  HREF="node104.html#SECTION001013100000000000000">Environment variables</A>
<LI><A ID="tex2html410"
  HREF="node104.html#SECTION001013200000000000000">Reading or writing files</A>
<LI><A ID="tex2html411"
  HREF="node104.html#SECTION001013300000000000000">Coordinate files and derivative data</A>
<LI><A ID="tex2html412"
  HREF="node104.html#SECTION001013400000000000000">Unicode</A>
</UL>
<LI><A ID="tex2html413"
  HREF="node105.html">File types</A>
</UL>
<LI><A ID="tex2html414"
  HREF="node106.html">Stereochemical parameters and molecular topology</A>
<UL>
<LI><A ID="tex2html415"
  HREF="node107.html">Modeling residues with non-existing or incomplete entries in the 
            topology and parameter libraries</A>
<UL>
<LI><A ID="tex2html416"
  HREF="node107.html#SECTION001021100000000000000">Residues with defined topology, but with missing 
parameters</A>
<LI><A ID="tex2html417"
  HREF="node107.html#SECTION001021200000000000000">Block (BLK) residues with undefined 
topology and parameters</A>
</UL>
</UL>
<LI><A ID="tex2html418"
  HREF="node108.html">Spatial restraints</A>
<UL>
<LI><A ID="tex2html419"
  HREF="node109.html">Specification of restraints</A>
<UL>
<LI><A ID="tex2html420"
  HREF="node109.html#SECTION001031100000000000000">Static and dynamic restraints</A>
<LI><A ID="tex2html421"
  HREF="node109.html#SECTION001031200000000000000">Feature types</A>
<LI><A ID="tex2html422"
  HREF="node109.html#SECTION001031300000000000000">Mathematical forms of restraints</A>
<LI><A ID="tex2html423"
  HREF="node109.html#SECTION001031400000000000000">Restraint violations</A>
</UL>
<LI><A ID="tex2html424"
  HREF="node110.html">Specification of pseudo atoms</A>
<LI><A ID="tex2html425"
  HREF="node111.html">Excluded pairs</A>
<LI><A ID="tex2html426"
  HREF="node112.html">Rigid bodies</A>
<LI><A ID="tex2html427"
  HREF="node113.html">Symmetry restraints</A>
</UL>
</UL>
<BR>
<LI><A ID="tex2html428"
  HREF="node114.html">M<SMALL>ODELLER</SMALL> command reference</A>
<UL>
<LI><A ID="tex2html429"
  HREF="node115.html">Key for command descriptions</A>
<LI><A ID="tex2html430"
  HREF="node116.html">The <TT>environ</TT> class: M<SMALL>ODELLER</SMALL> environment</A>
<UL>
<LI><A ID="tex2html431"
  HREF="node117.html">environ() &mdash; create a new M<SMALL>ODELLER</SMALL> environment</A>
<LI><A ID="tex2html432"
  HREF="node118.html">environ.io &mdash; default input parameters</A>
<LI><A ID="tex2html433"
  HREF="node119.html">environ.edat &mdash; default objective function parameters</A>
<LI><A ID="tex2html434"
  HREF="node120.html">environ.libs &mdash; M<SMALL>ODELLER</SMALL> libraries</A>
<LI><A ID="tex2html435"
  HREF="node121.html">environ.schedule_scale &mdash; energy function scaling factors</A>
<LI><A ID="tex2html436"
  HREF="node122.html">environ.dendrogram() &mdash; clustering</A>
<LI><A ID="tex2html437"
  HREF="node123.html">environ.principal_components() &mdash; clustering</A>
<LI><A ID="tex2html438"
  HREF="node124.html">environ.system() &mdash; execute system command</A>
<LI><A ID="tex2html439"
  HREF="node125.html">environ.make_pssmdb() &mdash; Create a database of PSSMs given a list of profiles</A>
</UL>
<LI><A ID="tex2html440"
  HREF="node126.html">The <TT>energy_data</TT> class: objective function parameters</A>
<UL>
<LI><A ID="tex2html441"
  HREF="node127.html">energy_data() &mdash; create a new set of objective function parameters</A>
<LI><A ID="tex2html442"
  HREF="node128.html">energy_data.contact_shell &mdash; nonbond distance cutoff</A>
<LI><A ID="tex2html443"
  HREF="node129.html">energy_data.update_dynamic &mdash; nonbond recalculation threshold</A>
<LI><A ID="tex2html444"
  HREF="node130.html">energy_data.sphere_stdv &mdash; soft-sphere standard deviation</A>
<LI><A ID="tex2html445"
  HREF="node131.html">energy_data.dynamic_sphere &mdash; calculate soft-sphere overlap restraints</A>
<LI><A ID="tex2html446"
  HREF="node132.html">energy_data.dynamic_lennard &mdash; calculate Lennard-Jones restraints</A>
<LI><A ID="tex2html447"
  HREF="node133.html">energy_data.dynamic_coulomb &mdash; calculate Coulomb restraints</A>
<LI><A ID="tex2html448"
  HREF="node134.html">energy_data.dynamic_modeller &mdash; calculate non-bonded spline restraints</A>
<LI><A ID="tex2html449"
  HREF="node135.html">energy_data.excl_local &mdash; exclude certain local pairs of atoms</A>
<LI><A ID="tex2html450"
  HREF="node136.html">energy_data.radii_factor &mdash; scale atomic radii</A>
<LI><A ID="tex2html451"
  HREF="node137.html">energy_data.lennard_jones_switch &mdash; Lennard-Jones switching parameters</A>
<LI><A ID="tex2html452"
  HREF="node138.html">energy_data.coulomb_switch &mdash; Coulomb switching parameters</A>
<LI><A ID="tex2html453"
  HREF="node139.html">energy_data.relative_dielectric &mdash; relative dielectric</A>
<LI><A ID="tex2html454"
  HREF="node140.html">energy_data.covalent_cys &mdash; use disulfide bridges in residue distance</A>
<LI><A ID="tex2html455"
  HREF="node141.html">energy_data.nonbonded_sel_atoms &mdash; control interaction with picked atoms</A>
<LI><A ID="tex2html456"
  HREF="node142.html">energy_data.nlogn_use &mdash; select non-bond list generation algorithm</A>
<LI><A ID="tex2html457"
  HREF="node143.html">energy_data.max_nlogn_grid_cells &mdash; maximum number of grid cells for NlogN nonbond pairs routine</A>
<LI><A ID="tex2html458"
  HREF="node144.html">energy_data.energy_terms &mdash; user-defined global energy terms</A>
</UL>
<LI><A ID="tex2html459"
  HREF="node145.html">The <TT>io_data</TT> class: coordinate file input parameters</A>
<UL>
<LI><A ID="tex2html460"
  HREF="node146.html">io_data() &mdash; create a new input parameters object</A>
<LI><A ID="tex2html461"
  HREF="node147.html">io_data.hetatm &mdash; whether to read HETATM records</A>
<LI><A ID="tex2html462"
  HREF="node148.html">io_data.hydrogen &mdash; whether to read hydrogen atoms</A>
<LI><A ID="tex2html463"
  HREF="node149.html">io_data.water &mdash; whether to read water molecules</A>
<LI><A ID="tex2html464"
  HREF="node150.html">io_data.convert_modres &mdash; whether to convert modified residues</A>
<LI><A ID="tex2html465"
  HREF="node151.html">io_data.atom_files_directory &mdash; search path for coordinate files</A>
</UL>
<LI><A ID="tex2html466"
  HREF="node152.html">The <TT>Libraries</TT> class: stereochemical parameters and molecular topology</A>
<UL>
<LI><A ID="tex2html467"
  HREF="node153.html">Libraries.topology &mdash; topology library information</A>
<LI><A ID="tex2html468"
  HREF="node154.html">Libraries.parameters &mdash; parameter library information</A>
<LI><A ID="tex2html469"
  HREF="node155.html">Topology.append() &mdash; append residue topology library</A>
<LI><A ID="tex2html470"
  HREF="node156.html">Topology.clear() &mdash; clear residue topology library</A>
<LI><A ID="tex2html471"
  HREF="node157.html">Topology.read() &mdash; read residue topology library</A>
<LI><A ID="tex2html472"
  HREF="node158.html">Parameters.append() &mdash; append parameters library</A>
<LI><A ID="tex2html473"
  HREF="node159.html">Parameters.clear() &mdash; clear parameters library</A>
<LI><A ID="tex2html474"
  HREF="node160.html">Parameters.read() &mdash; read parameters library</A>
<LI><A ID="tex2html475"
  HREF="node161.html">Topology.make() &mdash; make a subset topology library</A>
<LI><A ID="tex2html476"
  HREF="node162.html">Topology.submodel &mdash; select topology model type</A>
<LI><A ID="tex2html477"
  HREF="node163.html">Topology.write() &mdash; write residue topology library</A>
</UL>
<LI><A ID="tex2html478"
  HREF="node164.html">The <TT>model</TT> class: handling of atomic coordinates, and model building</A>
<UL>
<LI><A ID="tex2html479"
  HREF="node165.html">model() &mdash; create a new 3D model</A>
<LI><A ID="tex2html480"
  HREF="node166.html">model.seq_id &mdash; sequence identity between the model and templates</A>
<LI><A ID="tex2html481"
  HREF="node167.html">model.resolution &mdash; resolution of protein structure</A>
<LI><A ID="tex2html482"
  HREF="node168.html">model.last_energy &mdash; last objective function value</A>
<LI><A ID="tex2html483"
  HREF="node169.html">model.remark &mdash; text remark(s)</A>
<LI><A ID="tex2html484"
  HREF="node170.html">model.restraints &mdash; all static restraints which act on the model</A>
<LI><A ID="tex2html485"
  HREF="node171.html">model.group_restraints &mdash; all restraints which act on atom groups</A>
<LI><A ID="tex2html486"
  HREF="node172.html">model.atoms &mdash; all atoms in the model</A>
<LI><A ID="tex2html487"
  HREF="node173.html">model.point() &mdash; return a point in Cartesian space</A>
<LI><A ID="tex2html488"
  HREF="node174.html">model.atom_range() &mdash; return a subset of all atoms</A>
<LI><A ID="tex2html489"
  HREF="node175.html">model.residue_range() &mdash; return a subset of all residues</A>
<LI><A ID="tex2html490"
  HREF="node176.html">model.get_insertions() &mdash; return a list of all insertions</A>
<LI><A ID="tex2html491"
  HREF="node177.html">model.get_deletions() &mdash; return a list of all deletions</A>
<LI><A ID="tex2html492"
  HREF="node178.html">model.loops() &mdash; return a list of all loops</A>
<LI><A ID="tex2html493"
  HREF="node179.html">model.read() &mdash; read coordinates for MODEL</A>
<LI><A ID="tex2html494"
  HREF="node180.html">model.build_sequence() &mdash; build model from a sequence of one-letter codes</A>
<LI><A ID="tex2html495"
  HREF="node181.html">model.write() &mdash; write MODEL</A>
<LI><A ID="tex2html496"
  HREF="node182.html">model.clear_topology() &mdash; clear model topology</A>
<LI><A ID="tex2html497"
  HREF="node183.html">model.generate_topology() &mdash; generate MODEL topology</A>
<LI><A ID="tex2html498"
  HREF="node184.html">model.make_valid_pdb_coordinates() &mdash; make coordinates fit in PDB format</A>
<LI><A ID="tex2html499"
  HREF="node185.html">model.write_psf() &mdash; write molecular topology to PSF file</A>
<LI><A ID="tex2html500"
  HREF="node186.html">model.patch() &mdash; patch MODEL topology</A>
<LI><A ID="tex2html501"
  HREF="node187.html">model.patch_ss_templates() &mdash; guess MODEL disulfides from templates</A>
<LI><A ID="tex2html502"
  HREF="node188.html">model.patch_ss() &mdash; guess MODEL disulfides from model structure</A>
<LI><A ID="tex2html503"
  HREF="node189.html">model.build() &mdash; build MODEL coordinates from topology</A>
<LI><A ID="tex2html504"
  HREF="node190.html">model.transfer_xyz() &mdash; copy templates' coordinates to MODEL</A>
<LI><A ID="tex2html505"
  HREF="node191.html">model.res_num_from() &mdash; residue numbers from MODEL2 to MODEL</A>
<LI><A ID="tex2html506"
  HREF="node192.html">model.rename_segments() &mdash; rename MODEL segments</A>
<LI><A ID="tex2html507"
  HREF="node193.html">model.to_iupac() &mdash; standardize certain dihedral angles</A>
<LI><A ID="tex2html508"
  HREF="node194.html">model.reorder_atoms() &mdash; standardize order of MODEL atoms</A>
<LI><A ID="tex2html509"
  HREF="node195.html">model.orient() &mdash; center and orient MODEL</A>
<LI><A ID="tex2html510"
  HREF="node196.html">model.write_data() &mdash; write derivative model data</A>
<LI><A ID="tex2html511"
  HREF="node197.html">model.make_region() &mdash; define a random surface patch of atoms</A>
<LI><A ID="tex2html512"
  HREF="node198.html">model.color() &mdash; color MODEL according to alignment</A>
<LI><A ID="tex2html513"
  HREF="node199.html">model.make_chains() &mdash; Fetch sequences from PDB file</A>
<LI><A ID="tex2html514"
  HREF="node200.html">model.saxs_intens() &mdash; Calculate SAXS intensity from model</A>
<LI><A ID="tex2html515"
  HREF="node201.html">model.saxs_pr() &mdash; Calculate <SPAN CLASS="MATH"><IMG STYLE="height: 2.38ex; vertical-align: -0.70ex; " SRC="img1.png"
 ALT="$P(r)$"></SPAN> of model</A>
<LI><A ID="tex2html516"
  HREF="node202.html">model.saxs_chifun() &mdash; Calculate SAXS score chi from model</A>
<LI><A ID="tex2html517"
  HREF="node203.html">model.assess_ga341() &mdash; assess a model with the GA341 method</A>
<LI><A ID="tex2html518"
  HREF="node204.html">model.assess_normalized_dope() &mdash; assess a model with the normalized DOPE method</A>
<LI><A ID="tex2html519"
  HREF="node205.html">model.get_normalized_dope_profile() &mdash; get per-residue normalized DOPE profile</A>
</UL>
<LI><A ID="tex2html520"
  HREF="node206.html">The <TT>Restraints</TT> class: static restraints</A>
<UL>
<LI><A ID="tex2html521"
  HREF="node207.html">Restraints.rigid_bodies &mdash; all rigid bodies</A>
<LI><A ID="tex2html522"
  HREF="node208.html">Restraints.pseudo_atoms &mdash; all pseudo atoms</A>
<LI><A ID="tex2html523"
  HREF="node209.html">Restraints.excluded_pairs &mdash; all excluded pairs</A>
<LI><A ID="tex2html524"
  HREF="node210.html">Restraints.nonbonded_pairs &mdash; all nonbonded pairs</A>
<LI><A ID="tex2html525"
  HREF="node211.html">Restraints.symmetry &mdash; all symmetry restraints</A>
<LI><A ID="tex2html526"
  HREF="node212.html">Restraints.symmetry.report() &mdash; report violated symmetry restraints</A>
<LI><A ID="tex2html527"
  HREF="node213.html">Restraints.make() &mdash; make restraints</A>
<LI><A ID="tex2html528"
  HREF="node214.html">Restraints.make_distance() &mdash; make distance restraints</A>
<LI><A ID="tex2html529"
  HREF="node215.html">Restraints.unpick_all() &mdash; unselect all restraints</A>
<LI><A ID="tex2html530"
  HREF="node216.html">Restraints.clear() &mdash; delete all restraints</A>
<LI><A ID="tex2html531"
  HREF="node217.html">Restraints.pick() &mdash; pick restraints for selected atoms</A>
<LI><A ID="tex2html532"
  HREF="node218.html">Restraints.unpick_redundant() &mdash; unselect redundant restraints</A>
<LI><A ID="tex2html533"
  HREF="node219.html">Restraints.remove_unpicked() &mdash; remove unselected restraints</A>
<LI><A ID="tex2html534"
  HREF="node220.html">Restraints.condense() &mdash; remove unselected or redundant restraints</A>
<LI><A ID="tex2html535"
  HREF="node221.html">Restraints.add() &mdash; add restraint</A>
<LI><A ID="tex2html536"
  HREF="node222.html">Restraints.unpick() &mdash; unselect restraints</A>
<LI><A ID="tex2html537"
  HREF="node223.html">Restraints.reindex() &mdash; renumber model restraints using another model</A>
<LI><A ID="tex2html538"
  HREF="node224.html">Restraints.spline() &mdash; approximate restraints by splines</A>
<LI><A ID="tex2html539"
  HREF="node225.html">Restraints.append() &mdash; read spatial restraints</A>
<LI><A ID="tex2html540"
  HREF="node226.html">Restraints.write() &mdash; write spatial restraints</A>
</UL>
<LI><A ID="tex2html541"
  HREF="node227.html">The <TT>secondary_structure</TT> module: secondary structure restraints</A>
<UL>
<LI><A ID="tex2html542"
  HREF="node228.html">alpha() &mdash; make an <SPAN CLASS="MATH">&alpha;</SPAN>-helix</A>
<LI><A ID="tex2html543"
  HREF="node229.html">strand() &mdash; make a <SPAN CLASS="MATH">&beta;</SPAN>-strand</A>
<LI><A ID="tex2html544"
  HREF="node230.html">sheet() &mdash; make a <SPAN CLASS="MATH">&beta;</SPAN>-sheet</A>
</UL>
<LI><A ID="tex2html545"
  HREF="node231.html">The <TT>selection</TT> class: handling of sets of atom coordinates</A>
<UL>
<LI><A ID="tex2html546"
  HREF="node232.html">selection() &mdash; create a new selection</A>
<LI><A ID="tex2html547"
  HREF="node233.html">selection.add() &mdash; add objects to selection</A>
<LI><A ID="tex2html548"
  HREF="node234.html">selection.extend_by_residue() &mdash; extend selection by residue</A>
<LI><A ID="tex2html549"
  HREF="node235.html">selection.by_residue() &mdash; make sure all residues are fully selected</A>
<LI><A ID="tex2html550"
  HREF="node236.html">selection.select_sphere() &mdash; select all atoms within radius</A>
<LI><A ID="tex2html551"
  HREF="node237.html">selection.only_mainchain() &mdash; select only mainchain atoms</A>
<LI><A ID="tex2html552"
  HREF="node238.html">selection.only_sidechain() &mdash; select only sidechain atoms</A>
<LI><A ID="tex2html553"
  HREF="node239.html">selection.only_atom_types() &mdash; select only atoms of given types</A>
<LI><A ID="tex2html554"
  HREF="node240.html">selection.only_residue_types() &mdash; select only atoms of given residue type</A>
<LI><A ID="tex2html555"
  HREF="node241.html">selection.only_std_residues() &mdash; select only standard residues</A>
<LI><A ID="tex2html556"
  HREF="node242.html">selection.only_no_topology() &mdash; select only residues without topology</A>
<LI><A ID="tex2html557"
  HREF="node243.html">selection.only_het_residues() &mdash; select only HETATM residues</A>
<LI><A ID="tex2html558"
  HREF="node244.html">selection.only_water_residues() &mdash; select only water residues</A>
<LI><A ID="tex2html559"
  HREF="node245.html">selection.only_defined() &mdash; select only atoms with defined coordinates</A>
<LI><A ID="tex2html560"
  HREF="node246.html">selection.write() &mdash; write selection coordinates to a file</A>
<LI><A ID="tex2html561"
  HREF="node247.html">selection.translate() &mdash; translate all coordinates</A>
<LI><A ID="tex2html562"
  HREF="node248.html">selection.rotate_origin() &mdash; rotate coordinates about origin</A>
<LI><A ID="tex2html563"
  HREF="node249.html">selection.rotate_mass_center() &mdash; rotate coordinates about mass center</A>
<LI><A ID="tex2html564"
  HREF="node250.html">selection.transform() &mdash; transform coordinates with a matrix</A>
<LI><A ID="tex2html565"
  HREF="node251.html">selection.mutate() &mdash; mutate selected residues</A>
<LI><A ID="tex2html566"
  HREF="node252.html">selection.randomize_xyz() &mdash; randomize selected coordinates</A>
<LI><A ID="tex2html567"
  HREF="node253.html">selection.superpose() &mdash; superpose model on selection given alignment</A>
<LI><A ID="tex2html568"
  HREF="node254.html">selection.rotate_dihedrals() &mdash; change dihedral angles</A>
<LI><A ID="tex2html569"
  HREF="node255.html">selection.unbuild() &mdash; undefine coordinates</A>
<LI><A ID="tex2html570"
  HREF="node256.html">selection.hot_atoms() &mdash; atoms violating restraints</A>
<LI><A ID="tex2html571"
  HREF="node257.html">selection.energy() &mdash; evaluate atom selection given restraints</A>
<LI><A ID="tex2html572"
  HREF="node258.html">selection.debug_function() &mdash; test code self-consistency</A>
<LI><A ID="tex2html573"
  HREF="node259.html">selection.assess_dope() &mdash; assess a model selection with the DOPE method</A>
<LI><A ID="tex2html574"
  HREF="node260.html">selection.assess_dopehr() &mdash; assess a model with the DOPE-HR method</A>
<LI><A ID="tex2html575"
  HREF="node261.html">selection.get_dope_profile() &mdash; get per-residue DOPE profile</A>
<LI><A ID="tex2html576"
  HREF="node262.html">selection.get_dopehr_profile() &mdash; get per-residue DOPE-HR profile</A>
<LI><A ID="tex2html577"
  HREF="node263.html">selection.assess() &mdash; assess a model selection</A>
</UL>
<LI><A ID="tex2html578"
  HREF="node264.html">The <TT>physical</TT> module: contributions to the objective function</A>
<UL>
<LI><A ID="tex2html579"
  HREF="node265.html">physical.values() &mdash; create a new set of physical values</A>
</UL>
<LI><A ID="tex2html580"
  HREF="node266.html">The <TT>optimizers</TT> module: optimization of the model</A>
<UL>
<LI><A ID="tex2html581"
  HREF="node267.html">conjugate_gradients() &mdash; optimize atoms given restraints, with CG</A>
<LI><A ID="tex2html582"
  HREF="node268.html">quasi_newton() &mdash; optimize atoms with quasi-Newton minimization</A>
<LI><A ID="tex2html583"
  HREF="node269.html">molecular_dynamics() &mdash; optimize atoms given restraints, with MD</A>
<LI><A ID="tex2html584"
  HREF="node270.html">actions.write_structure() &mdash; write out the model coordinates</A>
<LI><A ID="tex2html585"
  HREF="node271.html">actions.trace() &mdash; write out optimization energies, <EM>etc</EM></A>
<LI><A ID="tex2html586"
  HREF="node272.html">actions.charmm_trajectory() &mdash; write out a CHARMM trajectory</A>
<LI><A ID="tex2html587"
  HREF="node273.html">User-defined optimizers</A>
</UL>
<LI><A ID="tex2html588"
  HREF="node274.html">The <TT>schedule</TT> class: variable target function optimization</A>
<UL>
<LI><A ID="tex2html589"
  HREF="node275.html">schedule() &mdash; create a new schedule</A>
<LI><A ID="tex2html590"
  HREF="node276.html">schedule.make_for_model() &mdash; trim a schedule for a model</A>
<LI><A ID="tex2html591"
  HREF="node277.html">schedule.write() &mdash; write optimization schedule</A>
</UL>
<LI><A ID="tex2html592"
  HREF="node278.html">The <TT>group_restraints</TT> class: restraints on atom groups</A>
<UL>
<LI><A ID="tex2html593"
  HREF="node279.html">group_restraints() &mdash; create a new set of group restraints</A>
<LI><A ID="tex2html594"
  HREF="node280.html">group_restraints.append() &mdash; read group restraint parameters</A>
</UL>
<LI><A ID="tex2html595"
  HREF="node281.html">The <TT>gbsa</TT> module: implicit solvation</A>
<UL>
<LI><A ID="tex2html596"
  HREF="node282.html">gbsa.Scorer() &mdash; create a new scorer to evaluate GB/SA energies</A>
</UL>
<LI><A ID="tex2html597"
  HREF="node283.html">SOAP potentials</A>
<UL>
<LI><A ID="tex2html598"
  HREF="node284.html">soap_loop.Scorer() &mdash; create a new scorer to evaluate SOAP-Loop energies</A>
<LI><A ID="tex2html599"
  HREF="node285.html">soap_peptide.Scorer() &mdash; create a new scorer to evaluate SOAP-Peptide energies</A>
<LI><A ID="tex2html600"
  HREF="node286.html">soap_pp.PairScorer() &mdash; create a new scorer to evaluate SOAP-PP pairwise energies</A>
<LI><A ID="tex2html601"
  HREF="node287.html">soap_pp.AtomScorer() &mdash; create a new scorer to evaluate SOAP-PP atomistic energies</A>
<LI><A ID="tex2html602"
  HREF="node288.html">soap_pp.Assessor() &mdash; assess with all components of the SOAP-PP score</A>
<LI><A ID="tex2html603"
  HREF="node289.html">soap_protein_od.Scorer() &mdash; create a new scorer to evaluate SOAP-Protein-OD energies</A>
</UL>
<LI><A ID="tex2html604"
  HREF="node290.html">The <TT>alignment</TT> class: comparison of sequences and structures</A>
<UL>
<LI><A ID="tex2html605"
  HREF="node291.html">alignment() &mdash; create a new alignment</A>
<LI><A ID="tex2html606"
  HREF="node292.html">alignment.comments &mdash; alignment file comments</A>
<LI><A ID="tex2html607"
  HREF="node293.html">alignment.positions &mdash; list of residue-residue alignment positions (including gaps)</A>
<LI><A ID="tex2html608"
  HREF="node294.html">alignment.append() &mdash; read sequences and/or their alignment</A>
<LI><A ID="tex2html609"
  HREF="node295.html">alignment.clear() &mdash; delete all sequences from the alignment</A>
<LI><A ID="tex2html610"
  HREF="node296.html">alignment.read_one() &mdash; read sequences one by one from a file</A>
<LI><A ID="tex2html611"
  HREF="node297.html">alignment.check_structure_structure() &mdash; check template structure superpositions</A>
<LI><A ID="tex2html612"
  HREF="node298.html">alignment.check_sequence_structure() &mdash; check sequence/structure alignment for sanity</A>
<LI><A ID="tex2html613"
  HREF="node299.html">alignment.check() &mdash; check alignment for modeling</A>
<LI><A ID="tex2html614"
  HREF="node300.html">alignment.compare_with() &mdash; compare two alignments</A>
<LI><A ID="tex2html615"
  HREF="node301.html">alignment.append_model() &mdash; copy model sequence and coordinates to alignment</A>
<LI><A ID="tex2html616"
  HREF="node302.html">alignment.append_sequence() &mdash; add a sequence from one-letter codes</A>
<LI><A ID="tex2html617"
  HREF="node303.html">alignment.append_profile() &mdash; add profile sequences to the alignment</A>
<LI><A ID="tex2html618"
  HREF="node304.html">alignment.write() &mdash; write sequences and/or their alignment</A>
<LI><A ID="tex2html619"
  HREF="node305.html">alignment.edit() &mdash; edit overhangs in alignment</A>
<LI><A ID="tex2html620"
  HREF="node306.html">alignment.describe() &mdash; describe proteins</A>
<LI><A ID="tex2html621"
  HREF="node307.html">alignment.id_table() &mdash; calculate percentage sequence identities</A>
<LI><A ID="tex2html622"
  HREF="node308.html">alignment.compare_sequences() &mdash; compare sequences in alignment</A>
<LI><A ID="tex2html623"
  HREF="node309.html">alignment.align() &mdash; align two (blocks of) sequences</A>
<LI><A ID="tex2html624"
  HREF="node310.html">alignment.align2d() &mdash; align sequences with structures</A>
<LI><A ID="tex2html625"
  HREF="node311.html">alignment.malign() &mdash; align two or more sequences</A>
<LI><A ID="tex2html626"
  HREF="node312.html">alignment.consensus() &mdash; consensus sequence alignment</A>
<LI><A ID="tex2html627"
  HREF="node313.html">alignment.compare_structures() &mdash; compare 3D structures given alignment</A>
<LI><A ID="tex2html628"
  HREF="node314.html">alignment.align3d() &mdash; align two structures</A>
<LI><A ID="tex2html629"
  HREF="node315.html">alignment.malign3d() &mdash; align two or more structures</A>
<LI><A ID="tex2html630"
  HREF="node316.html">alignment.salign() &mdash; align two or more sequences/structures of proteins</A>
<UL>
<LI><A ID="tex2html631"
  HREF="node316.html#SECTION00111626100000000000000">Features of proteins used for alignment</A>
<LI><A ID="tex2html632"
  HREF="node316.html#SECTION00111626200000000000000">Alignment of protein sequences</A>
<LI><A ID="tex2html633"
  HREF="node316.html#SECTION00111626300000000000000">Alignment of protein structures with sequences</A>
<LI><A ID="tex2html634"
  HREF="node316.html#SECTION00111626400000000000000">Alignment of protein structures</A>
<LI><A ID="tex2html635"
  HREF="node316.html#SECTION00111626500000000000000">Sub-optimal alignments</A>
<LI><A ID="tex2html636"
  HREF="node316.html#SECTION00111626600000000000000">Alignments using external restraints</A>
<LI><A ID="tex2html637"
  HREF="node316.html#SECTION00111626700000000000000">Gap penalties and correcting for gaps</A>
<LI><A ID="tex2html638"
  HREF="node316.html#SECTION00111626800000000000000">Useful SALIGN information and commands</A>
</UL>
<LI><A ID="tex2html639"
  HREF="node317.html">alignment.get_suboptimals() &mdash; parse suboptimal alignments file</A>
<LI><A ID="tex2html640"
  HREF="node318.html">alignment.to_profile() &mdash; convert alignment to profile format</A>
<LI><A ID="tex2html641"
  HREF="node319.html">alignment.segment_matching() &mdash; align segments</A>
</UL>
<LI><A ID="tex2html642"
  HREF="node320.html">The <TT>Sequence</TT> class: a single sequence within an alignment</A>
<UL>
<LI><A ID="tex2html643"
  HREF="node321.html">Sequence.range &mdash; residue range</A>
<LI><A ID="tex2html644"
  HREF="node322.html">Sequence.code &mdash; alignment code</A>
<LI><A ID="tex2html645"
  HREF="node323.html">Sequence.atom_file &mdash; PDB file name</A>
<LI><A ID="tex2html646"
  HREF="node324.html">Sequence.source &mdash; source organism</A>
<LI><A ID="tex2html647"
  HREF="node325.html">Sequence.name &mdash; protein name</A>
<LI><A ID="tex2html648"
  HREF="node326.html">Sequence.prottyp &mdash; protein sequence type</A>
<LI><A ID="tex2html649"
  HREF="node327.html">Sequence.resolution &mdash; structure resolution</A>
<LI><A ID="tex2html650"
  HREF="node328.html">Sequence.rfactor &mdash; R factor</A>
<LI><A ID="tex2html651"
  HREF="node329.html">Sequence.residues &mdash; list of all residues in the sequence</A>
<LI><A ID="tex2html652"
  HREF="node330.html">Sequence.chains &mdash; list of all chains in the sequence</A>
<LI><A ID="tex2html653"
  HREF="node331.html">Sequence.transfer_res_prop() &mdash; transfer residue properties</A>
<LI><A ID="tex2html654"
  HREF="node332.html">Sequence.get_num_equiv() &mdash; get number of equivalences</A>
<LI><A ID="tex2html655"
  HREF="node333.html">Sequence.get_sequence_identity() &mdash; get sequence identity</A>
</UL>
<LI><A ID="tex2html656"
  HREF="node334.html">The <TT>Structure</TT> class: a template structure within an alignment</A>
<UL>
<LI><A ID="tex2html657"
  HREF="node335.html">Structure.write() &mdash; write out PDB file</A>
<LI><A ID="tex2html658"
  HREF="node336.html">Structure.reread() &mdash; reread coordinates from the atom file</A>
<LI><A ID="tex2html659"
  HREF="node337.html">Structure.read() &mdash; read coordinates from a PDB file</A>
</UL>
<LI><A ID="tex2html660"
  HREF="node338.html">The <TT>Chain</TT> class: a single chain in a model or alignment</A>
<UL>
<LI><A ID="tex2html661"
  HREF="node339.html">Chain.name &mdash; chain ID</A>
<LI><A ID="tex2html662"
  HREF="node340.html">Chain.residues &mdash; all residues in the chain</A>
<LI><A ID="tex2html663"
  HREF="node341.html">Chain.atoms &mdash; all atoms in the chain</A>
<LI><A ID="tex2html664"
  HREF="node342.html">Chain.filter() &mdash; check if this chain passes all criteria</A>
<LI><A ID="tex2html665"
  HREF="node343.html">Chain.write() &mdash; write out chain sequence to an alignment file</A>
<LI><A ID="tex2html666"
  HREF="node344.html">Chain.atom_file_and_code() &mdash; get suitable names for this chain</A>
<LI><A ID="tex2html667"
  HREF="node345.html">Chain.join() &mdash; join other chain(s) onto this one</A>
</UL>
<LI><A ID="tex2html668"
  HREF="node346.html">The <TT>Residue</TT> class: a single residue in a model or alignment</A>
<UL>
<LI><A ID="tex2html669"
  HREF="node347.html">Residue.name &mdash; internal (CHARMM) residue type name</A>
<LI><A ID="tex2html670"
  HREF="node348.html">Residue.pdb_name &mdash; PDB (IUPAC) type name</A>
<LI><A ID="tex2html671"
  HREF="node349.html">Residue.code &mdash; One-letter residue type code</A>
<LI><A ID="tex2html672"
  HREF="node350.html">Residue.hetatm &mdash; HETATM indicator</A>
<LI><A ID="tex2html673"
  HREF="node351.html">Residue.index &mdash; internal integer index</A>
<LI><A ID="tex2html674"
  HREF="node352.html">Residue.num &mdash; PDB-style residue number</A>
<LI><A ID="tex2html675"
  HREF="node353.html">Residue.curvature &mdash; Mainchain curvature</A>
<LI><A ID="tex2html676"
  HREF="node354.html">Residue.atoms &mdash; all atoms in the residue</A>
<LI><A ID="tex2html677"
  HREF="node355.html">Residue.chain &mdash; chain object</A>
<LI><A ID="tex2html678"
  HREF="node356.html">Residue.phi &mdash; <SPAN CLASS="MATH">&phi;</SPAN> dihedral angle</A>
<LI><A ID="tex2html679"
  HREF="node357.html">Residue.psi &mdash; <SPAN CLASS="MATH">&psi;</SPAN> dihedral angle</A>
<LI><A ID="tex2html680"
  HREF="node358.html">Residue.omega &mdash; <SPAN CLASS="MATH">&omega;</SPAN> dihedral angle</A>
<LI><A ID="tex2html681"
  HREF="node359.html">Residue.alpha &mdash; <SPAN CLASS="MATH">&alpha;</SPAN> dihedral angle</A>
<LI><A ID="tex2html682"
  HREF="node360.html">Residue.chi1 &mdash; <SPAN CLASS="MATH">&chi;<sub>1</sub></SPAN> dihedral angle</A>
<LI><A ID="tex2html683"
  HREF="node361.html">Residue.chi2 &mdash; <SPAN CLASS="MATH">&chi;<sub>2</sub></SPAN> dihedral angle</A>
<LI><A ID="tex2html684"
  HREF="node362.html">Residue.chi3 &mdash; <SPAN CLASS="MATH">&chi;<sub>3</sub></SPAN> dihedral angle</A>
<LI><A ID="tex2html685"
  HREF="node363.html">Residue.chi4 &mdash; <SPAN CLASS="MATH">&chi;<sub>4</sub></SPAN> dihedral angle</A>
<LI><A ID="tex2html686"
  HREF="node364.html">Residue.chi5 &mdash; <SPAN CLASS="MATH">&chi;<sub>5</sub></SPAN> dihedral angle</A>
<LI><A ID="tex2html687"
  HREF="node365.html">Residue.get_aligned_residue() &mdash; get aligned residue in another sequence</A>
<LI><A ID="tex2html688"
  HREF="node366.html">Residue.add_leading_gaps() &mdash; add gap(s) before this residue</A>
<LI><A ID="tex2html689"
  HREF="node367.html">Residue.add_trailing_gaps() &mdash; add gap(s) after this residue</A>
<LI><A ID="tex2html690"
  HREF="node368.html">Residue.remove_leading_gaps() &mdash; remove gap(s) before this residue</A>
<LI><A ID="tex2html691"
  HREF="node369.html">Residue.remove_trailing_gaps() &mdash; remove gap(s) after this residue</A>
<LI><A ID="tex2html692"
  HREF="node370.html">Residue.get_leading_gaps() &mdash; get number of gaps before this residue</A>
<LI><A ID="tex2html693"
  HREF="node371.html">Residue.get_trailing_gaps() &mdash; get number of gaps after this residue</A>
</UL>
<LI><A ID="tex2html694"
  HREF="node372.html">The <TT>Dihedral</TT> class: a single dihedral in a model or alignment</A>
<UL>
<LI><A ID="tex2html695"
  HREF="node373.html">Dihedral.value &mdash; current value in degrees</A>
<LI><A ID="tex2html696"
  HREF="node374.html">Dihedral.atoms &mdash; atoms defining the angle</A>
<LI><A ID="tex2html697"
  HREF="node375.html">Dihedral.dihclass &mdash; integer dihedral class</A>
</UL>
<LI><A ID="tex2html698"
  HREF="node376.html">The <TT>Point</TT> class: a point in Cartesian space</A>
<UL>
<LI><A ID="tex2html699"
  HREF="node377.html">Point.x &mdash; x coordinate</A>
<LI><A ID="tex2html700"
  HREF="node378.html">Point.select_sphere() &mdash; select all atoms within radius</A>
</UL>
<LI><A ID="tex2html701"
  HREF="node379.html">The <TT>Atom</TT> class: a single atom in a model or structure</A>
<UL>
<LI><A ID="tex2html702"
  HREF="node380.html">Atom.dvx &mdash; objective function derivative</A>
<LI><A ID="tex2html703"
  HREF="node381.html">Atom.vx &mdash; x component of velocity</A>
<LI><A ID="tex2html704"
  HREF="node382.html">Atom.biso &mdash; isotropic temperature factor</A>
<LI><A ID="tex2html705"
  HREF="node383.html">Atom.accessibility &mdash; atomic accessibility</A>
<LI><A ID="tex2html706"
  HREF="node384.html">Atom.occ &mdash; occupancy</A>
<LI><A ID="tex2html707"
  HREF="node385.html">Atom.charge &mdash; electrostatic charge</A>
<LI><A ID="tex2html708"
  HREF="node386.html">Atom.mass &mdash; mass</A>
<LI><A ID="tex2html709"
  HREF="node387.html">Atom.name &mdash; PDB name</A>
<LI><A ID="tex2html710"
  HREF="node388.html">Atom.type &mdash; CHARMM atom type</A>
<LI><A ID="tex2html711"
  HREF="node389.html">Atom.residue &mdash; residue object</A>
<LI><A ID="tex2html712"
  HREF="node390.html">Atom.get_equivalent_atom() &mdash; get equivalent atom in another residue</A>
</UL>
<LI><A ID="tex2html713"
  HREF="node391.html">The <TT>AtomType</TT> class: a CHARMM atom type</A>
<UL>
<LI><A ID="tex2html714"
  HREF="node392.html">AtomType.name &mdash; CHARMM name</A>
<LI><A ID="tex2html715"
  HREF="node393.html">AtomType.mass &mdash; atomic mass</A>
<LI><A ID="tex2html716"
  HREF="node394.html">AtomType.element &mdash; element</A>
</UL>
<LI><A ID="tex2html717"
  HREF="node395.html">The <TT>EnergyProfile</TT> class: a per-residue energy profile</A>
<UL>
<LI><A ID="tex2html718"
  HREF="node396.html">EnergyProfile.min_rms &mdash; minimal RMS violation</A>
<LI><A ID="tex2html719"
  HREF="node397.html">EnergyProfile.heavy_rms &mdash; heavy RMS violation</A>
<LI><A ID="tex2html720"
  HREF="node398.html">EnergyProfile.get_normalized() &mdash; get a normalized energy profile</A>
<LI><A ID="tex2html721"
  HREF="node399.html">EnergyProfile.get_smoothed() &mdash; get a smoothed energy profile</A>
<LI><A ID="tex2html722"
  HREF="node400.html">EnergyProfile.write_to_file() &mdash; write to file</A>
</UL>
<LI><A ID="tex2html723"
  HREF="node401.html">The <TT>profile</TT> class: using sequence profiles</A>
<UL>
<LI><A ID="tex2html724"
  HREF="node402.html">profile() &mdash; create a new profile</A>
<LI><A ID="tex2html725"
  HREF="node403.html">profile.read() &mdash; read a profile of a sequence</A>
<LI><A ID="tex2html726"
  HREF="node404.html">profile.write() &mdash; write a profile</A>
<LI><A ID="tex2html727"
  HREF="node405.html">profile.to_alignment() &mdash; profile to alignment</A>
<LI><A ID="tex2html728"
  HREF="node406.html">profile.scan() &mdash; Compare a target profile against a database of profiles</A>
<LI><A ID="tex2html729"
  HREF="node407.html">profile.build() &mdash; Build a profile for a given sequence or alignment</A>
<LI><A ID="tex2html730"
  HREF="node408.html">pssmdb() &mdash; create a new PSSM database</A>
<LI><A ID="tex2html731"
  HREF="node409.html">pssmdb.read() &mdash; read a PSSM database from a file</A>
</UL>
<LI><A ID="tex2html732"
  HREF="node410.html">The <TT>sequence_db</TT> class: using sequence databases</A>
<UL>
<LI><A ID="tex2html733"
  HREF="node411.html">sequence_db() &mdash; create a new sequence database</A>
<LI><A ID="tex2html734"
  HREF="node412.html">sequence_db.read() &mdash; read a database of sequences</A>
<LI><A ID="tex2html735"
  HREF="node413.html">sequence_db.close() &mdash; close an open database</A>
<LI><A ID="tex2html736"
  HREF="node414.html">sequence_db.write() &mdash; write a database of sequences</A>
<LI><A ID="tex2html737"
  HREF="node415.html">sequence_db.convert() &mdash; convert a database to binary format</A>
<LI><A ID="tex2html738"
  HREF="node416.html">sequence_db.search() &mdash; search for similar sequences</A>
<LI><A ID="tex2html739"
  HREF="node417.html">sequence_db.filter() &mdash; cluster sequences by sequence-identity</A>
</UL>
<LI><A ID="tex2html740"
  HREF="node418.html">The <TT>density</TT> class: handling electron microscopy density data</A>
<UL>
<LI><A ID="tex2html741"
  HREF="node419.html">density() &mdash; create a new density map</A>
<LI><A ID="tex2html742"
  HREF="node420.html">density.resolution &mdash; Map resolution</A>
<LI><A ID="tex2html743"
  HREF="node421.html">density.sigma_factor &mdash; Sigma factor</A>
<LI><A ID="tex2html744"
  HREF="node422.html">density.voxel_size &mdash; Map voxel size</A>
<LI><A ID="tex2html745"
  HREF="node423.html">density.px &mdash; Origin of the map</A>
<LI><A ID="tex2html746"
  HREF="node424.html">density.py &mdash; Origin of the map</A>
<LI><A ID="tex2html747"
  HREF="node425.html">density.pz &mdash; Origin of the map</A>
<LI><A ID="tex2html748"
  HREF="node426.html">density.grid &mdash; Density values</A>
<LI><A ID="tex2html749"
  HREF="node427.html">density.read() &mdash; read an EM (electron microscopy) density map file</A>
<LI><A ID="tex2html750"
  HREF="node428.html">density.grid_search() &mdash; dock a structure into an EM (electron microscopy) density map</A>
</UL>
<LI><A ID="tex2html751"
  HREF="node429.html">The <TT>saxsdata</TT> class: using small-angle X-ray (SAXS) data</A>
<UL>
<LI><A ID="tex2html752"
  HREF="node430.html">saxsdata() &mdash; create a new saxsdata structure</A>
<LI><A ID="tex2html753"
  HREF="node431.html">saxsdata.ini_saxs() &mdash; Initialization of SAXS data</A>
<LI><A ID="tex2html754"
  HREF="node432.html">saxsdata.saxs_read() &mdash; Read in SAXS data</A>
<LI><A ID="tex2html755"
  HREF="node433.html">saxsdata.saxs_pr_read() &mdash; Read in P(r) data</A>
</UL>
<LI><A ID="tex2html756"
  HREF="node434.html">The <TT>info</TT> object: obtaining information about the M<SMALL>ODELLER</SMALL> build</A>
<UL>
<LI><A ID="tex2html757"
  HREF="node435.html">info.version &mdash; the full M<SMALL>ODELLER</SMALL> version number</A>
<LI><A ID="tex2html758"
  HREF="node436.html">info.version_info &mdash; the version number, as a tuple</A>
<LI><A ID="tex2html759"
  HREF="node437.html">info.build_date &mdash; the date this binary was built</A>
<LI><A ID="tex2html760"
  HREF="node438.html">info.exe_type &mdash; the executable type of this binary</A>
<LI><A ID="tex2html761"
  HREF="node439.html">info.debug &mdash; this binary's debug flag</A>
<LI><A ID="tex2html762"
  HREF="node440.html">info.bindir &mdash; M<SMALL>ODELLER</SMALL> binary directory</A>
<LI><A ID="tex2html763"
  HREF="node441.html">info.time_mark() &mdash; print current date, time, and CPU time</A>
<LI><A ID="tex2html764"
  HREF="node442.html">info.jobname &mdash; name of the current job</A>
</UL>
<LI><A ID="tex2html765"
  HREF="node443.html">The <TT>log</TT> object: controlling the amount of output</A>
<UL>
<LI><A ID="tex2html766"
  HREF="node444.html">log.level() &mdash; Set all log output levels</A>
<LI><A ID="tex2html767"
  HREF="node445.html">log.none() &mdash; display no log output</A>
<LI><A ID="tex2html768"
  HREF="node446.html">log.minimal() &mdash; display minimal log output</A>
<LI><A ID="tex2html769"
  HREF="node447.html">log.verbose() &mdash; display verbose log output</A>
<LI><A ID="tex2html770"
  HREF="node448.html">log.very_verbose() &mdash; display verbose log output, and dynamic memory information</A>
</UL>
<LI><A ID="tex2html771"
  HREF="node449.html">The <TT>modfile</TT> module: handling of files</A>
<UL>
<LI><A ID="tex2html772"
  HREF="node450.html">modfile.default() &mdash; generate an &lsquo;automatic&rsquo; filename</A>
<LI><A ID="tex2html773"
  HREF="node451.html">modfile.delete() &mdash; delete a file</A>
<LI><A ID="tex2html774"
  HREF="node452.html">modfile.inquire() &mdash; check if file exists</A>
<LI><A ID="tex2html775"
  HREF="node453.html">modfile.File() &mdash; open a handle to a M<SMALL>ODELLER</SMALL> file</A>
</UL>
<LI><A ID="tex2html776"
  HREF="node454.html">The <TT>scripts</TT> module: utility scripts</A>
<UL>
<LI><A ID="tex2html777"
  HREF="node455.html">cispeptide() &mdash; creates cis-peptide stereochemical restraints</A>
<LI><A ID="tex2html778"
  HREF="node456.html">complete_pdb() &mdash; read a PDB or mmCIF file, and fill in any missing atoms</A>
</UL>
<LI><A ID="tex2html779"
  HREF="node457.html">The <TT>salign</TT> module: high-level usage of SALIGN</A>
<UL>
<LI><A ID="tex2html780"
  HREF="node458.html">iterative_structural_align() &mdash; obtain the best structural alignment</A>
</UL>
<LI><A ID="tex2html781"
  HREF="node459.html">Parallel job support</A>
<UL>
<LI><A ID="tex2html782"
  HREF="node460.html">job() &mdash; create a new parallel job</A>
<LI><A ID="tex2html783"
  HREF="node461.html">sge_pe_job() &mdash; create a job using all Sun GridEngine (SGE) slave processes</A>
<LI><A ID="tex2html784"
  HREF="node462.html">sge_qsub_job() &mdash; create a job which can be expanded with Sun GridEngine 'qsub'</A>
<LI><A ID="tex2html785"
  HREF="node463.html">job.slave_startup_commands &mdash; Slave startup commands</A>
<LI><A ID="tex2html786"
  HREF="node464.html">job.queue_task() &mdash; submit a task to run within the job</A>
<LI><A ID="tex2html787"
  HREF="node465.html">job.run_all_tasks() &mdash; run all queued tasks, and return results</A>
<LI><A ID="tex2html788"
  HREF="node466.html">job.yield_tasks_unordered() &mdash; run all queued tasks, and yield unordered results</A>
<LI><A ID="tex2html789"
  HREF="node467.html">job.start() &mdash; start all slaves for message-passing</A>
<LI><A ID="tex2html790"
  HREF="node468.html">Communicator.send_data() &mdash; send data</A>
<LI><A ID="tex2html791"
  HREF="node469.html">Communicator.get_data() &mdash; get data</A>
<LI><A ID="tex2html792"
  HREF="node470.html">slave.run_cmd() &mdash; run a command on the slave</A>
<LI><A ID="tex2html793"
  HREF="node471.html">local_slave() &mdash; create a slave running on the local machine</A>
<LI><A ID="tex2html794"
  HREF="node472.html">sge_pe_slave() &mdash; create a slave running on a Sun GridEngine parallel environment slave node</A>
<LI><A ID="tex2html795"
  HREF="node473.html">sge_qsub_slave() &mdash; create a 'qsub' slave running on a Sun GridEngine node</A>
<LI><A ID="tex2html796"
  HREF="node474.html">ssh_slave() &mdash; create a slave on a remote host accessed via ssh</A>
</UL>
</UL>
<BR>
<LI><A ID="tex2html797"
  HREF="node475.html">M<SMALL>ODELLER</SMALL> low-level programming</A>
<UL>
<LI><A ID="tex2html798"
  HREF="node476.html">User-defined features and restraint forms</A>
<UL>
<LI><A ID="tex2html799"
  HREF="node477.html">User-defined feature types</A>
<LI><A ID="tex2html800"
  HREF="node478.html">User-defined restraint forms</A>
<LI><A ID="tex2html801"
  HREF="node479.html">User-defined energy terms</A>
</UL>
<LI><A ID="tex2html802"
  HREF="node480.html">M<SMALL>ODELLER</SMALL> programming interface (API)</A>
</UL>
<BR>
<LI><A ID="tex2html803"
  HREF="node481.html">Methods</A>
<UL>
<LI><A ID="tex2html804"
  HREF="node482.html">Dynamic programming for sequence and structure
             comparison and searching</A>
<UL>
<LI><A ID="tex2html805"
  HREF="node483.html">Pairwise comparison</A>
<LI><A ID="tex2html806"
  HREF="node484.html">Variable gap penalty</A>
<LI><A ID="tex2html807"
  HREF="node485.html">Local <EM>versus</EM> global alignment</A>
<LI><A ID="tex2html808"
  HREF="node486.html">Similarity <EM>versus</EM> distance scores</A>
<LI><A ID="tex2html809"
  HREF="node487.html">Multiple comparisons</A>
</UL>
<LI><A ID="tex2html810"
  HREF="node488.html">Optimization of the objective function by M<SMALL>ODELLER</SMALL></A>
<UL>
<LI><A ID="tex2html811"
  HREF="node489.html">Function</A>
<LI><A ID="tex2html812"
  HREF="node490.html">Optimizers</A>
<UL>
<LI><A ID="tex2html813"
  HREF="node490.html#SECTION001322100000000000000">Molecular dynamics</A>
<LI><A ID="tex2html814"
  HREF="node490.html#SECTION001322200000000000000">Langevin dynamics</A>
<LI><A ID="tex2html815"
  HREF="node490.html#SECTION001322300000000000000">Self-guided MD and LD</A>
<LI><A ID="tex2html816"
  HREF="node490.html#SECTION001322400000000000000">Rigid bodies</A>
<LI><A ID="tex2html817"
  HREF="node490.html#SECTION001322500000000000000">Rigid molecular dynamics</A>
<LI><A ID="tex2html818"
  HREF="node490.html#SECTION001322600000000000000">Rigid minimization</A>
</UL>
</UL>
<LI><A ID="tex2html819"
  HREF="node491.html">Equations used in the derivation of the molecular pdf</A>
<UL>
<LI><A ID="tex2html820"
  HREF="node492.html">Features and their derivatives</A>
<UL>
<LI><A ID="tex2html821"
  HREF="node492.html#SECTION001331100000000000000">Distance</A>
<LI><A ID="tex2html822"
  HREF="node492.html#SECTION001331200000000000000">Angle</A>
<LI><A ID="tex2html823"
  HREF="node492.html#SECTION001331300000000000000">Dihedral angle</A>
<LI><A ID="tex2html824"
  HREF="node492.html#SECTION001331400000000000000">Atomic solvent accessibility</A>
<LI><A ID="tex2html825"
  HREF="node492.html#SECTION001331500000000000000">Atomic density</A>
<LI><A ID="tex2html826"
  HREF="node492.html#SECTION001331600000000000000">Atomic coordinates</A>
</UL>
<LI><A ID="tex2html827"
  HREF="node493.html">Restraints and their derivatives</A>
<UL>
<LI><A ID="tex2html828"
  HREF="node493.html#SECTION001332100000000000000">Single Gaussian restraint</A>
<LI><A ID="tex2html829"
  HREF="node493.html#SECTION001332200000000000000">Multiple Gaussian restraint</A>
<LI><A ID="tex2html830"
  HREF="node493.html#SECTION001332300000000000000">Multiple binormal restraint</A>
<LI><A ID="tex2html831"
  HREF="node493.html#SECTION001332400000000000000">Lower bound</A>
<LI><A ID="tex2html832"
  HREF="node493.html#SECTION001332500000000000000">Upper bound</A>
<LI><A ID="tex2html833"
  HREF="node493.html#SECTION001332600000000000000">Cosine restraint</A>
<LI><A ID="tex2html834"
  HREF="node493.html#SECTION001332700000000000000">Coulomb restraint</A>
<LI><A ID="tex2html835"
  HREF="node493.html#SECTION001332800000000000000">Lennard-Jones restraint</A>
<LI><A ID="tex2html836"
  HREF="node493.html#SECTION001332900000000000000">Spline restraint</A>
<LI><A ID="tex2html837"
  HREF="node493.html#SECTION0013321000000000000000">Symmetry restraint</A>
</UL>
</UL>
<LI><A ID="tex2html838"
  HREF="node494.html">Flowchart of comparative modeling by M<SMALL>ODELLER</SMALL></A>
<LI><A ID="tex2html839"
  HREF="node495.html">Loop modeling method</A>
</UL>
<BR>
<LI><A ID="tex2html840"
  HREF="node496.html">File formats</A>
<UL>
<LI><A ID="tex2html841"
  HREF="node497.html">Alignment file (PIR)</A>
<LI><A ID="tex2html842"
  HREF="node498.html">Restraints file</A>
<UL>
<LI><A ID="tex2html843"
  HREF="node499.html">Restraints</A>
<LI><A ID="tex2html844"
  HREF="node500.html">Excluded pairs</A>
<LI><A ID="tex2html845"
  HREF="node501.html">Pseudo atoms</A>
<LI><A ID="tex2html846"
  HREF="node502.html">Symmetry restraints</A>
<LI><A ID="tex2html847"
  HREF="node503.html">Rigid bodies</A>
</UL>
<LI><A ID="tex2html848"
  HREF="node504.html">Profile file</A>
<LI><A ID="tex2html849"
  HREF="node505.html">Binary files</A>
</UL>
<BR>
<LI><A ID="tex2html850"
  HREF="node506.html">Converting T<SMALL>OP</SMALL> scripts from old M<SMALL>ODELLER</SMALL> versions</A>
<UL>
<LI><A ID="tex2html851"
  HREF="node507.html">Running old scripts unchanged</A>
<LI><A ID="tex2html852"
  HREF="node508.html">Converting T<SMALL>OP</SMALL> scripts to Python</A>
<UL>
<LI><A ID="tex2html853"
  HREF="node509.html">T<SMALL>OP</SMALL> commands and variables</A>
<LI><A ID="tex2html854"
  HREF="node510.html">T<SMALL>OP</SMALL> models and alignments</A>
<LI><A ID="tex2html855"
  HREF="node511.html">T<SMALL>OP</SMALL> to Python correspondence</A>
</UL>
</UL>
<BR>
<LI><A ID="tex2html856"
  HREF="node512.html">Bibliography</A>
<LI><A ID="tex2html857"
  HREF="node513.html">Index</A>
<LI><A ID="tex2html858"
  HREF="node514.html">About this document ...</A>
</UL>
<!--End of Table of Child-Links-->
<BR><HR>

</BODY>
</HTML>
