<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<!--Converted with LaTeX2HTML 2018.3 (Released July 19, 2018) -->
<HTML lang="EN">
<HEAD>
<TITLE>Restraints.make() &mdash; make restraints</TITLE>
<META NAME="description" CONTENT="Restraints.make() &mdash; make restraints">
<META NAME="keywords" CONTENT="manual">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=utf-8">
<META NAME="viewport" CONTENT="width=device-width, initial-scale=1.0">
<META NAME="Generator" CONTENT="LaTeX2HTML v2018.3">

<LINK REL="STYLESHEET" HREF="manual.css">
<LINK REL="STYLESHEET" HREF="pygments.css">

<LINK REL="next" HREF="node214.html">
<LINK REL="previous" HREF="node212.html">
<LINK REL="next" HREF="node214.html">
</HEAD>

<BODY >

<DIV CLASS="navigation"><!--Navigation Panel-->
<A
 HREF="node214.html">
<IMG WIDTH="37" HEIGHT="24" ALT="next" SRC="next.png"></A> 
<A
 HREF="node206.html">
<IMG WIDTH="26" HEIGHT="24" ALT="up" SRC="up.png"></A> 
<A
 HREF="node212.html">
<IMG WIDTH="63" HEIGHT="24" ALT="previous" SRC="prev.png"></A> 
<A ID="tex2html2914"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALT="contents" SRC="contents.png"></A> 
<A ID="tex2html2916"
  HREF="node513.html">
<IMG WIDTH="43" HEIGHT="24" ALT="index" SRC="index.png"></A> 
<BR>
<B> Next:</B> <A
 HREF="node214.html">Restraints.make_distance()   make</A>
<B> Up:</B> <A
 HREF="node206.html">The Restraints class: static</A>
<B> Previous:</B> <A
 HREF="node212.html">Restraints.symmetry.report()   report</A>
 &nbsp; <B>  <A ID="tex2html2915"
  HREF="node1.html">Contents</A></B> 
 &nbsp; <B>  <A ID="tex2html2917"
  HREF="node513.html">Index</A></B> 
<BR>
<BR></DIV>
<!--End of Navigation Panel-->

<H2><A ID="SECTION001177000000000000000">
Restraints.make() &mdash; make restraints</A>
</H2> <A NAME="18735"></A><A ID="CMD:Restraints.make"></A>
<A NAME="18738"></A><TT>make(atmsel, restraint_type, spline_on_site, residue_span_range=(0, 99999), residue_span_sign=True, restraint_sel_atoms=1, basis_pdf_weight='LOCAL', basis_relative_weight=0.05, intersegment=True, dih_lib_only=False, spline_dx=0.5, spline_min_points=5, spline_range=4.0, mnch_lib=1, accessibility_type=8, surftyp=1, distngh=6.0, aln=None, edat=None, io=None)</TT>
<BR>
<DL>
<DT><STRONG>Requirements:</STRONG></DT>
<DD>topology <TT>&amp;</TT> parameters
                             
</DD>
</DL>

<P>
<BLOCKQUOTE>
This command calculates and selects new restraints of
a specified type. See the original papers for the most detailed definition
and description of the restraints [<A
 HREF="node512.html#SalBlu93">&#352;ali &amp; Blundell, 1993</A>,<A
 HREF="node512.html#SalOve94">&#352;ali &amp; Overington, 1994</A>]. The
calculation of restraints of all types is now (partly) limited to the 
atoms in the atom selection <I CLASS="sans">atmsel</I><A NAME="18921"></A>. The new restraints are added to any
currently present.
</BLOCKQUOTE>
<P>
<BLOCKQUOTE>The physical restraint type of the new restraints is specified by
<I CLASS="sans">restraint_group</I><A NAME="18924"></A>, and should be an object from the <TT>physical</TT>
module (see Table&nbsp;<A HREF="node265.html#tab:physrsrtypes">6.1</A>).
</BLOCKQUOTE>
<P>
<BLOCKQUOTE><I CLASS="sans">restraint_type</I><A NAME="18929"></A> selects the types of the generated restraints. 
(For restraint type <TT>DISTANCE</TT>, do not use this
command; instead, use <B><A HREF="node214.html#CMD:Restraints.makedistance">Restraints.make_distance()</A></B><A NAME="18933"></A>.)
Only one restraint type can be selected at a time, except
for the stereochemical restraints (<TT>BOND</TT>, <TT>ANGLE</TT>, <TT>DIHEDRAL</TT>, 
<TT>IMPROPER</TT>)
that can all be calculated at the same time. It is useful to
distinguish between the stereochemical restraints and homology-derived
restraints. The stereochemical restraints are obtained from 
libraries that depend on atom and/or residue types only (<EM>e.g.</EM>, 
C<SMALL>HARMM</SMALL>&nbsp;22 force field [<A
 HREF="node512.html#CHARMM98">MacKerell <EM>et&nbsp;al.</EM>, 1998</A>] or statistical 
potentials), and do not require the alignment <I CLASS="sans">aln</I><A NAME="18944"></A> with template structures.
In contrast, the homology-derived 
restraints are calculated from related protein structures, which
correspond to all but the last sequence in the alignment <I CLASS="sans">aln</I><A NAME="18947"></A> (the target).
These templates are read from coordinate files,
which are the only data files required. All restraints are added to the
existing restraints, even if they duplicate them (but see the comment
for the <TT>'OMEGA'</TT> restraints below).
</BLOCKQUOTE>
<P>
<BLOCKQUOTE><A ID="SECTION:span"></A> The atoms for non-bonded restraints also have to be
within the residue spanning range specified by
<I CLASS="sans">residue_span_range</I><A NAME="18951"></A> = <TT>(r1, r2)</TT>, such that the residue index
difference <!-- MATH
 $r1 \le |ir2 - ir1| \le r2$
 -->
<SPAN CLASS="MATH"><IMG STYLE="height: 2.38ex; vertical-align: -0.70ex; " SRC="img63.png"
 ALT="$r1 \le \vert ir2 - ir1\vert \le r2$"></SPAN> when <I CLASS="sans">residue_span_sign</I><A NAME="18955"></A> =
<TT>False</TT> and <!-- MATH
 $r1 \le (ir2 - ir1) \le r2$
 -->
<SPAN CLASS="MATH"><IMG STYLE="height: 2.38ex; vertical-align: -0.70ex; " SRC="img64.png"
 ALT="$r1 \le (ir2 - ir1) \le r2$"></SPAN> when <I CLASS="sans">residue_span_sign</I><A NAME="18959"></A> =
<TT>True</TT>.
</BLOCKQUOTE>
<P>
<BLOCKQUOTE><B>Stereochemical restraints:</B>
</BLOCKQUOTE>
<P>

<UL>
<LI><TT>'BOND'</TT>. This calculates covalent bond restraints (harmonic terms). 
It relies on the 
list of the atom-atom bonds for MODEL, prepared previously by the 
<B><A HREF="node183.html#CMD:model.generatetopology">model.generate_topology()</A></B><A NAME="18964"></A> command. The mean values and force constants are 
obtained from the parameter library in memory. Only those bonds
are restrained that have all or at least <I CLASS="sans">restraint_sel_atoms</I><A NAME="18969"></A>
in the selection <I CLASS="sans">atmsel</I><A NAME="18972"></A>.

<P>
</LI>
<LI><TT>'ANGLE'</TT>. This calculates covalent angle restraints (harmonic
terms). It relies on the
list of the atom-atom-atom bonds for MODEL, prepared previously by the 
<B><A HREF="node183.html#CMD:model.generatetopology">model.generate_topology()</A></B><A NAME="18976"></A> command. The mean values and force constants are 
obtained from the parameter library in memory. Only those angles
are restrained that have all or at least <I CLASS="sans">restraint_sel_atoms</I><A NAME="18981"></A>
in the selection <I CLASS="sans">atmsel</I><A NAME="18984"></A>.

<P>
</LI>
<LI><TT>'DIHEDRAL'</TT>. This calculates covalent dihedral angle restraints 
(cosine terms). It relies on the list of the atom-atom-atom-atom 
dihedral angles for MODEL, prepared previously by the <B><A HREF="node183.html#CMD:model.generatetopology">model.generate_topology()</A></B><A NAME="18988"></A> 
command. The minima, phases, and force constants are obtained from the 
parameter library in memory. Only those dihedral angles are restrained 
that have all or at least <I CLASS="sans">restraint_sel_atoms</I><A NAME="18993"></A> in the selection
<I CLASS="sans">atmsel</I><A NAME="18996"></A>.

<P>
</LI>
<LI><TT>'IMPROPER'</TT>. This calculates improper dihedral angle restraints
(harmonic terms). It relies on the list of the improper dihedral angles 
for MODEL, prepared previously by the 
<B><A HREF="node183.html#CMD:model.generatetopology">model.generate_topology()</A></B><A NAME="19000"></A> command. The mean values and force constants are 
obtained from the parameter library in memory. Only those impropers
are restrained that have all or at least <I CLASS="sans">restraint_sel_atoms</I><A NAME="19005"></A>
in the selection <I CLASS="sans">atmsel</I><A NAME="19008"></A>.

<P>
</LI>
<LI><TT>'STEREO'</TT>. This implies all <TT>'BOND'</TT>, <TT>'ANGLE'</TT>, <TT>'DIHEDRAL'</TT>, and 
<TT>'IMPROPER'</TT> restraints.

<P>
</LI>
<LI><TT>'SPHERE14'</TT>. This constructs soft-sphere<A NAME="19016"></A> overlap restraints (lower 
harmonic bounds) for atom pairs separated by exactly three bonds (1-4 
pairs). It relies on atom radii from the <TT>'$RADII14_LIB'</TT> library.
Only those non-bonded pairs are restrained that have all or at least 
<I CLASS="sans"><A HREF="node141.html#MEMB:energydata.nonbondedselatoms">energy_data.nonbonded_sel_atoms</A></I><A NAME="19019"></A> in the selection <I CLASS="sans">atmsel</I><A NAME="19024"></A>.
They must also satisfy the <I CLASS="sans">residue_span_range</I><A NAME="19027"></A> &amp; <I CLASS="sans">residue_span_sign</I><A NAME="19030"></A>
criterion.

<P>
</LI>
<LI><TT>'LJ14'</TT>. This constructs 1-4 Lennard-Jones<A NAME="19033"></A> restraints using the 
modified 1-4 Lennard-Jones<A NAME="19034"></A> parameters from the C<SMALL>HARMM</SMALL> parameter library.
There is no way to calculate <TT>'LJ14'</TT> as dynamic restraints.
Only those non-bonded pairs are restrained that have all or at least 
<I CLASS="sans"><A HREF="node141.html#MEMB:energydata.nonbondedselatoms">energy_data.nonbonded_sel_atoms</A></I><A NAME="19038"></A> in the selection <I CLASS="sans">atmsel</I><A NAME="19043"></A>.
They must also satisfy the <I CLASS="sans">residue_span_range</I><A NAME="19046"></A> &amp; <I CLASS="sans">residue_span_sign</I><A NAME="19049"></A>
criterion.

<P>
</LI>
<LI><TT>'COULOMB14'</TT>. This constructs 1-4 Coulomb<A NAME="19052"></A> restraints by relying
on the atomic charges from the C<SMALL>HARMM</SMALL> topology library.
There is no way to calculate <TT>'COULOMB14'</TT> as dynamic restraints.
Only those non-bonded pairs are restrained that have all or at least 
<I CLASS="sans"><A HREF="node141.html#MEMB:energydata.nonbondedselatoms">energy_data.nonbonded_sel_atoms</A></I><A NAME="19056"></A> in the selection <I CLASS="sans">atmsel</I><A NAME="19061"></A>.
They must also satisfy the <I CLASS="sans">residue_span_range</I><A NAME="19064"></A> &amp; <I CLASS="sans">residue_span_sign</I><A NAME="19067"></A>
criterion.

<P>
</LI>
<LI><TT>'SPHERE'</TT>. This constructs soft-sphere<A NAME="19070"></A> overlap restraints
(lower harmonic bounds) for all atom pairs that are not in bonds, 
angles, dihedral angles, improper dihedral angles, nor are explicitly 
excluded by the <TT>'E'</TT> entries read from a restraint file or added
by the <B><A HREF="node221.html#CMD:Restraints.add">Restraints.add()</A></B><A NAME="19073"></A> command. 
Only those non-bonded pairs are restrained that have all or at least 
<I CLASS="sans"><A HREF="node141.html#MEMB:energydata.nonbondedselatoms">energy_data.nonbonded_sel_atoms</A></I><A NAME="19078"></A> in the selection <I CLASS="sans">atmsel</I><A NAME="19083"></A>.
They must also satisfy the <I CLASS="sans">residue_span_range</I><A NAME="19086"></A> &amp; <I CLASS="sans">residue_span_sign</I><A NAME="19089"></A>
criterion.  Note that this makes these restraints 
static (<EM>i.e.</EM>, not dynamic) and that you must set <I CLASS="sans"><A HREF="node131.html#MEMB:energydata.dynamicsphere">energy_data.dynamic_sphere</A></I><A NAME="19093"></A>
to <TT>False</TT> before evaluating the molecular pdf if you want to avoid 
duplicated restraints. These restraints should usually not be combined 
with the Lennard-Jones<A NAME="19098"></A> (<TT>'LJ'</TT>) restraints.

<P>
When <I CLASS="sans">intersegment</I><A NAME="19101"></A> is <TT>True</TT>, the inter-segment non-bonded restraints 
are also constructed; otherwise, the segments do not feel each other 
<EM>via</EM> the non-bonded restraints. This option does not apply to the
optimizers (Section&nbsp;<A HREF="node266.html#SECTION:optimizers">6.11</A>) where information about segments
is not used at all (<EM>i.e.</EM>, they behave as if <I CLASS="sans">intersegment</I><A NAME="19107"></A> = <TT>True</TT>).

<P>
</LI>
<LI><TT>'LJ'</TT>. This constructs Lennard-Jones<A NAME="19111"></A> restraints for all atom pairs 
that are not in bonds, angles, dihedral angles, improper dihedral angles, 
nor are explicitly excluded by the <TT>'E'</TT> entries read from a restraint file 
or added by the <B><A HREF="node221.html#CMD:Restraints.add">Restraints.add()</A></B><A NAME="19114"></A> command. 
Only those non-bonded pairs are restrained that have all or at least 
<I CLASS="sans"><A HREF="node141.html#MEMB:energydata.nonbondedselatoms">energy_data.nonbonded_sel_atoms</A></I><A NAME="19119"></A> in the selection <I CLASS="sans">atmsel</I><A NAME="19124"></A>.
They must also satisfy the <I CLASS="sans">residue_span_range</I><A NAME="19127"></A> &amp; <I CLASS="sans">residue_span_sign</I><A NAME="19130"></A>
criterion.
Note that this command makes the non-bonded
restraints static (<EM>i.e.</EM>, not dynamic) and that you must set
<I CLASS="sans"><A HREF="node132.html#MEMB:energydata.dynamiclennard">energy_data.dynamic_lennard</A></I><A NAME="19134"></A>
to <TT>False</TT> before evaluating the molecular pdf if you want to avoid 
duplicated restraints. Note that C<SMALL>HARMM</SMALL> uses both <TT>'LJ14'</TT> and <TT>'LJ'</TT>.
For large molecules, it is better to calculate <TT>'LJ'</TT> as dynamic restraints 
because you can use distance cutoff <I CLASS="sans"><A HREF="node128.html#MEMB:energydata.contactshell">energy_data.contact_shell</A></I><A NAME="19144"></A> in
optimization (Section&nbsp;<A HREF="node266.html#SECTION:optimizers">6.11</A>) to
reduce significantly the number of non-bonded atom pairs.

<P>
</LI>
<LI><TT>'COULOMB'</TT>. This constructs Coulomb<A NAME="19149"></A> restraints for all atom pairs that 
are not in bonds, angles, dihedral angles, improper dihedral angles, nor are 
explicitly excluded by the <TT>'E'</TT> entries read from a restraint file or added
by the <B><A HREF="node221.html#CMD:Restraints.add">Restraints.add()</A></B><A NAME="19152"></A> command. 
Only those non-bonded pairs are restrained that have all or at least 
<I CLASS="sans"><A HREF="node141.html#MEMB:energydata.nonbondedselatoms">energy_data.nonbonded_sel_atoms</A></I><A NAME="19157"></A> in the selection <I CLASS="sans">atmsel</I><A NAME="19162"></A>.
They must also satisfy the <I CLASS="sans">residue_span_range</I><A NAME="19165"></A> &amp; <I CLASS="sans">residue_span_sign</I><A NAME="19168"></A>
criterion.
Note that this command makes the non-bonded restraints 
static (<EM>i.e.</EM>, not dynamic) and that you must set
<I CLASS="sans"><A HREF="node133.html#MEMB:energydata.dynamiccoulomb">energy_data.dynamic_coulomb</A></I><A NAME="19172"></A>
to <TT>False</TT> before evaluating the molecular pdf if you want to avoid 
duplicated restraints. Note that C<SMALL>HARMM</SMALL> uses both <TT>'COULOMB14'</TT> and 
<TT>'COULOMB'</TT>. For large molecules, it is better to calculate <TT>'COULOMB'</TT> 
as dynamic restraints because you can use distance cutoff
<I CLASS="sans"><A HREF="node128.html#MEMB:energydata.contactshell">energy_data.contact_shell</A></I><A NAME="19182"></A> in optimization
(Section&nbsp;<A HREF="node266.html#SECTION:optimizers">6.11</A>)
to reduce significantly the number of non-bonded atom pairs.

<P>
</LI>
</UL>
<P>
<BLOCKQUOTE><B>Homology-derived restraints:</B>
</BLOCKQUOTE>
<P>
<BLOCKQUOTE>For these restraints, the input alignment <I CLASS="sans">aln</I><A NAME="19187"></A> must be given.
</BLOCKQUOTE>
<P>

<UL>
<LI><TT>'CHI1_DIHEDRAL'</TT>, <TT>'CHI2_DIHEDRAL'</TT>,
<TT>'CHI3_DIHEDRAL'</TT>, <TT>'CHI4_DIHEDRAL'</TT>, <TT>'PHI_DIHEDRAL'</TT>, <TT>'PSI_DIHEDRAL'</TT>,
<TT>'OMEGA_DIHEDRAL'</TT>, <TT>'PHI-PSI_BINORMAL'</TT> are the mainchain and sidechain 
dihedral angle restraints. Only those dihedral angles are restrained that 
have all or at least <I CLASS="sans"><A HREF="node141.html#MEMB:energydata.nonbondedselatoms">energy_data.nonbonded_sel_atoms</A></I><A NAME="19198"></A> in the selection
<I CLASS="sans">atmsel</I><A NAME="19203"></A>. The means and standard deviations for the dihedral 
Gaussian restraints are obtained from the <TT>$RESDIH_LIB</TT> and 
<TT>$MNCH?_LIB</TT> libraries and their weights from the MDT tables,
which are read in as specified by <TT>MDT_LIB</TT> in <TT>$LIB/libs.lib</TT>.
The large MDT tables give the conditional weights for each possible dihedral
angle class, as a function of all possible combinations of features on which 
a particular class depends. If <I CLASS="sans">dih_lib_only</I><A NAME="19209"></A> is <TT>True</TT> or there is 
no equivalent residue in any of the
templates, the weights for the dihedral angle classes depend only on
the residue type and are obtained from the <TT>'$RESDIH_LIB'</TT> and 
<TT>'$MNCH?_LIB'</TT> libraries; the <I CLASS="sans">dih_lib_only</I><A NAME="19215"></A> argument allows
one to force the calculation of the &ldquo;homology-derived&rdquo; mainchain
and sidechain dihedral angle restraints that ignore template information.
<I CLASS="sans">basis_pdf_weight</I><A NAME="19218"></A> has the same effect as 
for the distance pdf's.

<P>
When M<SMALL>ODELLER</SMALL>'s <TT>'OMEGA'</TT> restraints are calculated, the currently existing 
restraints on atoms <TT>'O C +N +CA'</TT> in all residues are automatically deleted. 
These deleted restraints correspond to the <I>improper</I> dihedral angles
involving the <SPAN CLASS="MATH">&omega;</SPAN> atoms. They are deleted because they could be 
&ldquo;frustrated&rdquo; by the new <TT>'OMEGA'</TT> restraints. No action is taken with 
regard to any of the previously existing, possibly duplicated dihedral angle
restraints. Thus, to avoid restraint duplication, including that of the
<TT>'OMEGA'</TT> restraints, call the <B><A HREF="node218.html#CMD:Restraints.unpickredundant">Restraints.unpick_redundant()</A></B><A NAME="19227"></A> command
after all the restraints are calculated.

<P>
Dihedral restraints are only calculated for the 20 standard amino acids,
plus the two alternate histidine protonation states (HSE and HSP). The
statistics for HIS are applied to HSE and HSP.
</LI>
</UL>
<P>
<BLOCKQUOTE>The weights of basis pdf's depend on local sequence similarity between
the target and the templates when <I CLASS="sans">basis_pdf_weight</I><A NAME="19232"></A> = <TT>'LOCAL'</TT>
and on global sequence identity when <I CLASS="sans">basis_pdf_weight</I><A NAME="19236"></A> = <TT>'GLOBAL'</TT>.
</BLOCKQUOTE>
<P>
<BLOCKQUOTE><I CLASS="sans">basis_relative_weight</I><A NAME="19240"></A> is the cutoff for removing weak basis pdf's
from poly-Gaussian feature pdf's: a basis pdf whose weight is less
than the <I CLASS="sans">basis_relative_weight</I><A NAME="19243"></A> fraction of the largest weight is deleted.
</BLOCKQUOTE>
<P>
<BLOCKQUOTE>If <I CLASS="sans">spline_on_site</I><A NAME="19246"></A> is <TT>True</TT>, then certain dihedral restraints are
automatically replaced by splines for efficiency. See <B><A HREF="node224.html#CMD:Restraints.spline">Restraints.spline()</A></B><A NAME="19250"></A>
for a description of the <I CLASS="sans">spline_dx</I><A NAME="19255"></A>, <I CLASS="sans">spline_min_points</I><A NAME="19258"></A>, and
<I CLASS="sans">spline_range</I><A NAME="19261"></A> parameters.
</BLOCKQUOTE>
<P>
<BLOCKQUOTE>Several restraint types look up information from pre-calculated M<SMALL>DT</SMALL> tables,
and for these the <I CLASS="sans">accessibility_type</I><A NAME="19265"></A> variable defines the type of
solvent accessibility.

</BLOCKQUOTE>

<P>

  <DL>
<DT><STRONG>Example: <A ID="tex2html122"
  HREF="../examples/commands/make_restraints.py">examples/commands/make_restraints.py</A></STRONG></DT>
<DD> <BR>  <div class="pygments"><pre><span></span><span class="c1"># Example for: restraints.make(), restraints.spline(), restraints.write()</span>

<span class="c1"># This will compare energies of bond length restraints expressed</span>
<span class="c1"># by harmonic potential and by cubic spline.</span>

<span class="kn">from</span> <span class="nn">modeller</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">modeller.scripts</span> <span class="kn">import</span> <span class="n">complete_pdb</span>

<span class="n">log</span><span class="o">.</span><span class="n">verbose</span><span class="p">()</span>
<span class="n">env</span> <span class="o">=</span> <span class="n">environ</span><span class="p">()</span>
<span class="n">env</span><span class="o">.</span><span class="n">io</span><span class="o">.</span><span class="n">atom_files_directory</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;../atom_files&#39;</span><span class="p">]</span>
<span class="n">env</span><span class="o">.</span><span class="n">libs</span><span class="o">.</span><span class="n">topology</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="nb">file</span><span class="o">=</span><span class="s1">&#39;$(LIB)/top_heav.lib&#39;</span><span class="p">)</span>
<span class="n">env</span><span class="o">.</span><span class="n">libs</span><span class="o">.</span><span class="n">parameters</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="nb">file</span><span class="o">=</span><span class="s1">&#39;$(LIB)/par.lib&#39;</span><span class="p">)</span>

<span class="n">code</span> <span class="o">=</span> <span class="s1">&#39;1fas&#39;</span>
<span class="n">mdl</span> <span class="o">=</span> <span class="n">complete_pdb</span><span class="p">(</span><span class="n">env</span><span class="p">,</span> <span class="n">code</span><span class="p">)</span>
<span class="n">mdl</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="nb">file</span><span class="o">=</span><span class="n">code</span><span class="o">+</span><span class="s1">&#39;.ini&#39;</span><span class="p">)</span>

<span class="n">sel</span> <span class="o">=</span> <span class="n">selection</span><span class="p">(</span><span class="n">mdl</span><span class="p">)</span>
<span class="n">mdl</span><span class="o">.</span><span class="n">restraints</span><span class="o">.</span><span class="n">make</span><span class="p">(</span><span class="n">sel</span><span class="p">,</span> <span class="n">restraint_type</span><span class="o">=</span><span class="s1">&#39;bond&#39;</span><span class="p">,</span> <span class="n">spline_on_site</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">mdl</span><span class="o">.</span><span class="n">restraints</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="nb">file</span><span class="o">=</span><span class="n">code</span><span class="o">+</span><span class="s1">&#39;-1.rsr&#39;</span><span class="p">)</span>
<span class="n">edat</span> <span class="o">=</span> <span class="n">energy_data</span><span class="p">(</span><span class="n">dynamic_sphere</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">sel</span><span class="o">.</span><span class="n">energy</span><span class="p">(</span><span class="n">edat</span><span class="o">=</span><span class="n">edat</span><span class="p">)</span>

<span class="n">mdl</span><span class="o">.</span><span class="n">restraints</span><span class="o">.</span><span class="n">spline</span><span class="p">(</span><span class="n">forms</span><span class="o">.</span><span class="n">gaussian</span><span class="p">,</span> <span class="n">features</span><span class="o">.</span><span class="n">distance</span><span class="p">,</span> <span class="n">physical</span><span class="o">.</span><span class="n">bond</span><span class="p">,</span>
                      <span class="n">spline_range</span><span class="o">=</span><span class="mf">5.0</span><span class="p">,</span> <span class="n">spline_dx</span><span class="o">=</span><span class="mf">0.005</span><span class="p">,</span> <span class="n">edat</span><span class="o">=</span><span class="n">edat</span><span class="p">)</span>
<span class="n">mdl</span><span class="o">.</span><span class="n">restraints</span><span class="o">.</span><span class="n">condense</span><span class="p">()</span>
<span class="n">mdl</span><span class="o">.</span><span class="n">restraints</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="nb">file</span><span class="o">=</span><span class="n">code</span><span class="o">+</span><span class="s1">&#39;-2.rsr&#39;</span><span class="p">)</span>
<span class="n">sel</span><span class="o">.</span><span class="n">energy</span><span class="p">(</span><span class="n">edat</span><span class="o">=</span><span class="n">edat</span><span class="p">)</span>
</pre></div>
  
</DD>
</DL>  <BR>
<P>

<DIV CLASS="navigation"><HR>
<!--Navigation Panel-->
<A
 HREF="node214.html">
<IMG WIDTH="37" HEIGHT="24" ALT="next" SRC="next.png"></A> 
<A
 HREF="node206.html">
<IMG WIDTH="26" HEIGHT="24" ALT="up" SRC="up.png"></A> 
<A
 HREF="node212.html">
<IMG WIDTH="63" HEIGHT="24" ALT="previous" SRC="prev.png"></A> 
<A ID="tex2html2914"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALT="contents" SRC="contents.png"></A> 
<A ID="tex2html2916"
  HREF="node513.html">
<IMG WIDTH="43" HEIGHT="24" ALT="index" SRC="index.png"></A> 
<BR>
<B> Next:</B> <A
 HREF="node214.html">Restraints.make_distance()   make</A>
<B> Up:</B> <A
 HREF="node206.html">The Restraints class: static</A>
<B> Previous:</B> <A
 HREF="node212.html">Restraints.symmetry.report()   report</A>
 &nbsp; <B>  <A ID="tex2html2915"
  HREF="node1.html">Contents</A></B> 
 &nbsp; <B>  <A ID="tex2html2917"
  HREF="node513.html">Index</A></B> </DIV>
<!--End of Navigation Panel-->

</BODY>
</HTML>
